app/src/main/kotlin/com/example/android/picover/presentation/navigation/PicoverNavigationBar.kt

```diff
@@ -1,36 +1,61 @@
 package com.example.android.picover.presentation.navigation
 
+import androidx.activity.compose.BackHandler
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material3.DrawerState
+import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.material3.Icon
+import androidx.compose.material3.ModalNavigationDrawer
 import androidx.compose.material3.NavigationBar
 import androidx.compose.material3.NavigationBarItem
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.currentBackStackEntryAsState
+import com.example.android.picover.presentation.ProfileDrawerContent
+import com.example.android.picover.presentation.navigation.NavigationItem.Profile
+import kotlinx.coroutines.launch
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun PicoverNavigationBar(
 	items: List<NavigationItem>,
 	navController: NavHostController,
+	drawerState: DrawerState,
 	modifier: Modifier = Modifier,
 	onItemClick: (NavigationItem) -> Unit
 ) {
 	val backStackEntry = navController.currentBackStackEntryAsState()
 
 	Column {
-		PicoverNavHost(
+		ModalNavigationDrawer(
+			drawerContent = { ProfileDrawerContent() },
+			drawerState = drawerState,
+			gesturesEnabled = drawerState.isOpen,
 			modifier = modifier.weight(1f),
-			navController = navController,
-			startDestination = NavigationItem.Home.route
-		)
+		) {
+			val coroutineScope = rememberCoroutineScope()
+			BackHandler(enabled = drawerState.isOpen) {
+				coroutineScope.launch { drawerState.close() }
+			}
+			PicoverNavHost(
+				modifier = modifier.weight(1f),
+				navController = navController,
+				startDestination = NavigationItem.Home.route
+			)
+		}
 		NavigationBar(modifier = Modifier.fillMaxWidth()) {
 			items.forEach { item ->
 				NavigationBarItem(
-					selected = item.route == backStackEntry.value?.destination?.route,
+					selected = if (item is Profile) {
```

Comment:
also here: extract such logic like this or it is better to have this like here :D

Reply:
For me, if it’s only UI related it can stay as is.